src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -174,10 +174,6 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
 
             private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
             {
-                // First see if this is a document/project removal.  If so, clear out any state we
-                // have associated with any analyzers we have for that document/project.
-                ProcessRemovedDiagnostics(e);
```

Comment:
@mavasani no that is not the issue. issue is that the aggregation tagger holds onto diagnostics by its id in diagnostic event args. and that id is only unique per workspace.

now 2 different workspaces, preview and host workspaces raised diagnostic updated events that happen to use same diagnostic event id.

and the aggregate tagger didn't filter out events that's not belong to itself.